### PR TITLE
Panther Alien

### PR DESCRIPTION
**Pull request by drawsstuff**
## About The Pull Request

Adds the first of many Toy Aliens (If you don't count the bull). The panther alien is a slower, less damaging runner, with more health and tackle damage to compensate. It's unique lunge stuns targets for twice as long, but suffers a much longer cool down to compensate. Aimed at being the best at 1 on 1 host capturing, the panther alien is intended for fans of the runner caste looking to spice things up!

## Why It's Good For The Game

“Toy Aliens” have been a personal side project of mine to offer alternative playstyles for fans of a certain caste. Each of them are based off of an existing Xenomorph toy unless stated otherwise. The Toy alien project aims to give xeno players modified existing castes to play with instead of "Strains" or "Mutators". 

You can find more about the Toy Alien project here: https://docs.google.com/document/d/1CMz6aC4E6jLfDBzayhGfrZOvfovwNh9Cd2GkCNwDosE/edit?usp=sharing

🆑
add: Adds the Panther Alien, a specialized runner.
/:cl:

### CODE REVIEW

**Comment by David-Stormwell** [comment]
It is really good at what it does really, even better at the defiler as MJP commented with 50 tackle, I'd myself suggest making it a T2? Or prehaps putting a soft cap on how many can exist.

**Comment by drawsstuff** [comment]
just wok e up, pr has some issues mjp foudn, ivv be tweasking the numbers, hold tight on testi merging this please

**Comment by drawsstuff** [comment]
Alright, speed is gonna be smoothed a little, but ancient value will stay the same.

File: code/modules/mob/living/carbon/xenomorph/castes/panther/castedatum_panther.dm [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	evolves_to = list(/mob/living/carbon/xenomorph/hunter, /mob/living/carbon/xenomorph/bull, /mob/living/carbon/xenomorph/runner)
+
+	// *** Flags *** //
+	caste_flags = CASTE_CAN_BE_QUEEN_HEALED|CASTE_EVOLUTION_ALLOWED|CASTE_CAN_VENT_CRAWL|CASTE_CAN_BE_GIVEN_PLASMA|CASTE_CAN_BE_LEADER
+
+	// *** Defense *** //
+	armor = list("melee" = 10, "bullet" = 10, "laser" = 10, "energy" = 5, "bomb" = XENO_BOMB_RESIST_0, "bio" = 0, "rad" = 0, "fire" = 0, "acid" = 0)
+
+	// *** Ranged Attack *** //
+	charge_type = 3 //Heavy Pounce - Panther
```
Comment (Fikou) [inline code/modules/mob/living/carbon/xenomorph/castes/panther/castedatum_panther.dm:45]: shouldnt this be CHARGE_TYPE_PANTHER? defines exist for a reason

File: code/modules/mob/living/carbon/xenomorph/castes/panther/castedatum_panther.dm [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+	// *** Ranged Attack *** //
+	pounce_delay = 45.0 SECONDS
+
+/datum/xeno_caste/panther/ancient
+	upgrade_name = "Ancient"
+	caste_desc = "The colony only falls silent when a predator is a on the prowl."
+	ancient_message = "We are the apex predator, all will fall by fang or claw."
+	upgrade = XENO_UPGRADE_THREE
+	wound_type = "runner" //used to match appropriate wound overlays
```
Comment (Fikou) [inline code/modules/mob/living/carbon/xenomorph/castes/panther/castedatum_panther.dm:130]: unneeded i think

**Comment by jroinc** [comment]
Ah yess, the PANTHER!

Let's discuss this dumpster-fire of an alien.
Intended role-Ambusher and stamina damager.

Speed is low at young, but improves rapidly to be acceptable.
Health is low, but it's a T1 which is acceptable.
Armor's low-ish, but again, T1.
Melee damage is so low that it can't harm armored humans (intended).

It has one pounce and hiding ability, which allows it to stun a human for 4.5 seconds (supposedly).
And it has very high tackle damage.

Here are some scenarios that happened last round.

Scout spec alone, I'm hidden. I pounce him and get off 6 tackles. He goes to orange health and shotguns me, and I have to run. ZERO DAMAGE INFLICTED.

Critically hurt PFC, I'm hidden, spitter as ally. I pounce him, both of us spam-tackle, 7 of mine, 2-3 of spitter's. He gets up and WALKS to the other ungas, just out of sight and I have to run. ZERO DAMAGE INFLICTED.

One PFC, unknown armor, I'm hidden. Six tackles, he shotguns me, I run. ZERO DAMAGE INFLICTED.

One nested medic, IS armor (told through LOOC). I have to unnest him to move him. ELEVEN tackles later, he drops. He had time to get out of the pounce and knife me 3 times. If he had a shotgun, I'd be dead. ZERO DAMAGE INFLICTED.

Same nested medic, second breakout timer. 10-12 tackles, he drops. He had sufficient time to OD and die. ZERO DAMAGE INFLICTED. HOST LOST.

Nested pyro. 6 tackles in, he flames the panther. 12 tackles or so in, the flamer drops. The panther resists ONCE. The flamer gets back up and flames him again. Panther dies. ZERO DAMAGE INFLICTED. HOST LOST. MUTUAL KILL~

Three panthers died that round. Collectively, we managed to capture ONE guy, and that's cause he got double-teamed by 2 panthers.

ADDITIONAL PROBLEM-

He has one ability. I'm reasonably certain that I could give up plasma recovery for a whole match and have no issues whatsoever.

POSSIBLE SOLUTIONS-

Grant the panther neuroclaws/neurosting.

6 tackles will give 30u neurotox, which is actually reasonable in that they'll staminacrit on their own if alone and a medic doesn't get to them. They can self-treat, but this causes some toxin damage over time.

Neurosting is the same, but requires greater skill to pull off, and removes risk of OD.

File: code/modules/mob/living/carbon/xenomorph/castes/panther/castedatum_panther.dm (hunk 2 above)
Comment (drawsstuff) [reply code/modules/mob/living/carbon/xenomorph/castes/panther/castedatum_panther.dm:130]: All ancients have a special message upon reaching ancient, along with a new description,

Comment (Fikou) [reply code/modules/mob/living/carbon/xenomorph/castes/panther/castedatum_panther.dm:130]: this was a comment on wound_type

File: code/modules/mob/living/carbon/xenomorph/castes/panther/castedatum_panther.dm [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	// *** Melee Attacks *** //
+	melee_damage = 10
+	attack_delay = 6
+
+
+	// *** Tackle *** //
+	tackle_damage = 33
+
+	// *** Speed *** //
+	speed = .5
```
Comment (psykzz) [inline code/modules/mob/living/carbon/xenomorph/castes/panther/castedatum_panther.dm:23]:
```suggestion
	speed = 0.5
```

**Comment by drawsstuff** [comment]
https://streamable.com/81qh11

I gave my own caste a spin. I think it's fine, personally. It's not as a good as a runner, but it's also not supposed to be a runner.

**Comment by jroinc** [comment]
> https://streamable.com/81qh11
> I gave my own caste a spin. I think it's fine, personally. It's not as a good as a runner, but it's also not supposed to be a runner.

In all 4 of those cases, you didn't get the capture, it was you and 2-3 other ayys combining multiple stun abilities. Is that the intended use?

**Comment by drawsstuff** [comment]
> > https://streamable.com/81qh11
> > I gave my own caste a spin. I think it's fine, personally. It's not as a good as a runner, but it's also not supposed to be a runner.
> 
> In all 4 of those cases, you didn't get the capture, it was you and 2-3 other ayys combining multiple stun abilities. Is that the intended use?

I dunno, it looks like they were captured to me, infact, in all 4, it looks like I was the one dragging them off in the end. 

This is a T1, and a T1 isnt gonna get far on its own, no xeno will. But with some teamwork, you can go above and beyond.

**Comment by MemedHams** [comment]
"switching" from panther to runner via Evo eats a T2 slot. Also, though I agree T1's aren't supposed to be particularly powerful, this one is literally, utterly useless. Consider a buff or adding some neuro.

File: code/modules/mob/living/carbon/xenomorph/xenoprocs.dm [hunk 4]
```diff
@@ -327,7 +327,10 @@
 
 					visible_message("<span class='danger'>[src] pounces on [M]!</span>",
 									"<span class='xenodanger'>We pounce on [M]!</span>", null, 5)
-					M.Paralyze(20)
+					if(xeno_caste.charge_type == CHARGE_TYPE_PANTHER)
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:330]: please dont do this, this code is already pretty bad.

Comment (drawsstuff) [reply code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:330]: What should I replace it with? This is my first time coding. Would M.Paralyze((xeno_caste.charge_type == CHARGE_TYPE_PANTHER) ? 45 : 20) be just as bad?

Comment (spookydonut) [reply code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:330]: special casing the charge type inside a ranged switch case to just change the stun timer is not great.

Comment (spookydonut) [reply code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:330]: ideally this would be determined by the pounce action and not hard coded like this

Comment (spookydonut) [reply code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:330]: i can do this for you since you're new

Comment (spookydonut) [reply code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:330]: You know what this is fine as-is i'll just merge this and pr a refactor of this whole messy proc

Comment (drawsstuff) [reply code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:330]: Thank you

**Pull request merged**